genesis/engine/solvers/rigid/collider_decomp.py

```diff
@@ -50,6 +51,42 @@ def rotmatx(matin, i0, i1, i2, f0, f1, f2):
     return matres
 
 
+@ti.func
+def check_collision_valid(
+    links_state: array_class.LinksState,
+    links_info: array_class.LinksInfo,
+    geoms_info: array_class.GeomsInfo,
+    static_rigid_sim_config: ti.template(),
+    collision_pair_validity: ti.template(),
+    i_ga: int,
+    i_gb: int,
+    i_b: int,
+):
+    is_valid = collision_pair_validity[i_ga, i_gb]
+
+    # hibernated <-> fixed links
+    if ti.static(static_rigid_sim_config.use_hibernation):
+        i_la = geoms_info[i_ga].link_idx
+        i_lb = geoms_info[i_gb].link_idx
+        I_la = [i_la, i_b] if ti.static(static_rigid_sim_config.batch_links_info) else i_la
+        I_lb = [i_lb, i_b] if ti.static(static_rigid_sim_config.batch_links_info) else i_lb
+
+        if (links_state[i_la, i_b].hibernated and links_info[I_lb].is_fixed) or (
+            links_state[i_lb, i_b].hibernated and links_info[I_la].is_fixed
+        ):
+            is_valid = False
+
+    return is_valid
+
+
+@ti.func
+def is_geom_aabbs_overlap(geoms_state: array_class.GeomsState, i_ga: int, i_gb: int, i_b: int):
```

Comment:
can you keep it as a class method instead of a standalone function to minimize code change?